eng/pipelines/mono/templates/workloads-build.yml

```diff
@@ -0,0 +1,100 @@
+  dependsOn: []
+  isOfficialBuild: false
+  osGroup: ''
+  osSubgroup: ''
+  platform: ''
+  pool: ''
+  runtimeVariant: ''
+  stagedBuild: false
+  testGroup: ''
+  timeoutInMinutes: ''
+  variables: {}
+
+jobs:
+- template: xplat-pipeline-job.yml
+  parameters:
+    archType: ${{ parameters.archType }}
+    buildConfig: ${{ parameters.buildConfig }}
+    container: ${{ parameters.container }}
+    condition: ${{ parameters.isOfficialBuild }}
+    helixType: 'build/product/'
+    osGroup: ${{ parameters.osGroup }}
+    osSubgroup: ${{ parameters.osSubgroup }}
+    pool: ${{ parameters.pool }}
+    runtimeVariant: ${{ parameters.runtimeVariant }}
+    stagedBuild: ${{ parameters.stagedBuild }}
+    timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+    dependOnEvaluatePaths: ${{ parameters.dependOnEvaluatePaths }}
+
+    dependsOn: ${{ parameters.dependsOn }}
+
+    name: workloadsbuild
+    displayName: Build Workloads
+
+    variables:
+    - name: officialBuildIdArg
+      value: ''
+    - ${{ if and(eq(variables['System.TeamProject'], 'internal'), ne(variables['Build.Reason'], 'PullRequest')) }}:
+      - name: officialBuildIdArg
+        value: '/p:OfficialBuildId=$(Build.BuildNumber)'
+    - name: SignType
+      value: $[ coalesce(variables.OfficialSignType, 'real') ]
+    - ${{ parameters.variables }}
+
+    steps:
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifact: 'IntermediateArtifacts'
+        path: $(workloadPackagesPath)
+        patterns: |
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.browser-wasm*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.android-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.browser-wasm*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.ios-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.iossimulator-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.maccatalyst-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.tvos-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NETCore.App.Runtime.Mono.tvossimulator-*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NET.Workload.Mono.ToolChain.Manifest*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NET.Runtime.MonoTargets.Sdk*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NET.Runtime.MonoAOTCompiler.Task*.nupkg
+          IntermediateArtifacts/MonoRuntimePacks/Shipping/Microsoft.NET.Runtime.WebAssembly.Sdk*.nupkg
```

Comment:
Can you think of a way to avoid hardcoding all these packages? Would creating a new intermediate artifact only for the mono runtime packs make sense?

Reply:
The purpose of the hardcoded list is to speed up builds, by only downloading the _required_ artifacts out of the build. Just grabbing the whole MonoRuntimePacks folder weighs in at around 4GiB